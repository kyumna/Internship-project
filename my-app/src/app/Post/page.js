"use client"
import Link from 'next/link';
import React, { useEffect, useState } from 'react';


export default function Home() {

  const [users, setUsers] = useState([]);

  useEffect(() => {
    const fetchUsers = async () => {
      try {
        const response = await fetch('https://jsonplaceholder.typicode.com/posts');
        const data = await response.json();
        setUsers(data);
      } catch (error) {
        console.log('Error fetching data:', error);
      }
    };

    fetchUsers();
  }, []);

  return (
    <div className='bg-pastelPink'>
    <Link href="/" className='float-right mr-2 pt-2 font-bold'>See Users</Link>
   
      <h1 className="text-center  text-2xl font-bold pb-2 pt-2 ">Posts</h1>
      {users.map((user) => (
        <div key={user.id} className='w-46 p-2 m-4 border-1 border-gray-300 shadow-xl  rounded-2xl bg-white cursor-pointer hover:transform hover:scale-105 transition-all duration-300'>
         
          <div className='block'><span className='font-bold'>Title</span>: {user.title}</div>
          <div className='block'><span className='font-bold'>Body</span>: {user.body}</div>
        </div>
      ))}
    </div>
  );
}

