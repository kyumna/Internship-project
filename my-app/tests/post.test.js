import React from "react";
import { render } from "react-dom";
import Post from '../src/app/Post/page'
import { act } from "react-dom/test-utils";
 const MOCK_DATA = [
  {
    "id": 1,
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
  },
];
let container;

// beforeEach function runs before each 
// of the tests in this file run to setup a testing environment  
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

// afterEach runs after each one of the tests in this file completes 
// to teardown the testing environment
afterEach(() => {
  document.body.removeChild(container);
  container = null;
});
describe("Posts", () => {
    it("Renders without crashing", async () => {
    
    // This part fakes fetch and resolves it by return MOCK_DATA
    global.fetch = jest.fn().mockImplementation(() =>
        Promise.resolve({
          json: () => Promise.resolve(MOCK_DATA),
        })
      );
      
      // This part renders a component inside of container that we created before
      await act(async () => {
        render(<Post />, container);
      });
  
      expect(container).toMatchSnapshot();
    });
  });