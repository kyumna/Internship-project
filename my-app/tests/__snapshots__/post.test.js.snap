// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Renders without crashing 1`] = `
<div>
  <div
    class="bg-pastelPink"
  >
    <a
      class="float-right mr-2 pt-2 font-bold"
      href="/"
    >
      See Users
    </a>
    <h1
      class="text-center  text-2xl font-bold pb-2 pt-2 "
    >
      Posts
    </h1>
    <div
      class="w-46 p-2 m-4 border-1 border-gray-300 shadow-xl  rounded-2xl bg-white cursor-pointer hover:transform hover:scale-105 transition-all duration-300"
      data-testid="post"
    >
      <div
        class="block"
      >
        <span
          class="font-bold"
        >
          Title
        </span>
        : 
        sunt aut facere repellat provident occaecati excepturi optio reprehenderit
      </div>
      <div
        class="block"
      >
        <span
          class="font-bold"
        >
          Body
        </span>
        : 
        quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
      </div>
    </div>
  </div>
</div>
`;
